src/Sonarr.Http/Frontend/InitializeJsonController.cs

```diff
@@ -16,18 +19,28 @@ public class InitializeJsonController : Controller
         private readonly IConfigFileProvider _configFileProvider;
         private readonly IAnalyticsService _analyticsService;
 
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        private readonly IUserService _userService;
+
         private static string _apiKey;
         private static string _urlBase;
         private string _generatedContent;
 
         public InitializeJsonController(IConfigFileProvider configFileProvider,
-                                      IAnalyticsService analyticsService)
+                                      IAnalyticsService analyticsService,
+                                      IHttpContextAccessor httpContextAccessor,
+                                      IUserService userService)
         {
             _configFileProvider = configFileProvider;
             _analyticsService = analyticsService;
-
-            _apiKey = configFileProvider.ApiKey;
             _urlBase = configFileProvider.UrlBase;
+
+            _httpContextAccessor = httpContextAccessor;
+            _userService = userService;
+
+            var currentUser = GetCurrentUser();
+            _apiKey = currentUser != null ? currentUser.ApiKey : "None";
```

Comment:
Is "None" the right value to use here, or would null be better? (Is this just a display property perhaps?)

Reply:
I don't think this is needed, it's only going to be called once (singleton) and instead should be looked up during the request.